src/index.ts

```diff
@@ -5,10 +5,19 @@ export async function renderWith(
   require: (dep: string) => any,
   storyResult: unknown,
   div: HTMLElement
-): Promise<boolean | VoidFunction> {
+): Promise<void> {
   const storyType = typeOf(storyResult);
-  const rendered = await render(require, storyResult, storyType, div);
-  if (!rendered) {
+  const dispose = await render(require, storyResult, storyType, div);
+
+  if (typeof dispose === 'function') {
+    const win = div.ownerDocument.defaultView;
+    win.addEventListener("unload", dispose);
+    
+    const observer = new MutationObserver(dispose);
+    observer.observe(div.parentNode, {childList: true});
```

Comment:
just thinking you do not need to filter children to avoid unmoutimg the wrong one ?

Reply:
I believe the div#root is the only child of body, but I'll filter this as a safety mechanism for the future.

Reply:
yes today but if I have let say an overlay inside the same div than the story ... everything will be unmounted I think no ?

Reply:
Exactly, adding some sibling could cause an ugly bug here. Better filtering it to avoid that.